From defb3f540a0ad41b324fb0e23b24778545be6f75 Mon Sep 17 00:00:00 2001
From: sebhz <seb@zyglute.fr>
Date: Mon, 7 Nov 2022 19:53:40 +0100
Subject: [PATCH 2/3] Personal config


diff --git a/config.def.h b/config.def.h
index 2e9276a..195e322 100644
--- a/config.def.h
+++ b/config.def.h
@@ -3,37 +3,44 @@
 /* appearance */
 static const unsigned int borderpx  = 1;        /* border pixel of windows */
 static const unsigned int snap      = 32;       /* snap pixel */
+
 static const unsigned int systraypinning = 0;   /* 0: sloppy systray follows selected monitor, >0: pin systray to monitor X */
-static const unsigned int systrayonleft = 0;   	/* 0: systray in the right corner, >0: systray on left of status text */
+static const unsigned int systrayonleft = 0;    /* 0: systray in the right corner, >0: systray on left of status text */
 static const unsigned int systrayspacing = 2;   /* systray spacing */
 static const int systraypinningfailfirst = 1;   /* 1: if pinning fails, display systray on the first monitor, False: display systray on the last monitor*/
-static const int showsystray        = 1;     /* 0 means no systray */
+static const int showsystray        = 1;        /* 0 means no systray */
+
 static const int showbar            = 1;        /* 0 means no bar */
 static const int topbar             = 1;        /* 0 means bottom bar */
-static const char *fonts[]          = { "monospace:size=10" };
-static const char dmenufont[]       = "monospace:size=10";
+static const char *fonts[]          = { "monospace:size=9" };
+static const char dmenufont[]       = "monospace:size=9";
 static const char col_gray1[]       = "#222222";
 static const char col_gray2[]       = "#444444";
 static const char col_gray3[]       = "#bbbbbb";
 static const char col_gray4[]       = "#eeeeee";
 static const char col_cyan[]        = "#005577";
+static const char col_emerald[]     = "#005c46";
+static const char col_red[]         = "#ff0000";
+
 static const char *colors[][3]      = {
 	/*               fg         bg         border   */
 	[SchemeNorm] = { col_gray3, col_gray1, col_gray2 },
-	[SchemeSel]  = { col_gray4, col_cyan,  col_cyan  },
+	[SchemeSel]  = { col_gray4, col_emerald,  col_red },
 };
 
 /* tagging */
-static const char *tags[] = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+static const char *tags[] = { "1", "2", "3", "4" };
 
 static const Rule rules[] = {
 	/* xprop(1):
 	 *	WM_CLASS(STRING) = instance, class
 	 *	WM_NAME(STRING) = title
 	 */
-	/* class      instance    title       tags mask     isfloating   monitor */
-	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	/* class                        instance    title       tags mask     isfloating   monitor */
+	{ "Gimp",                       NULL,       NULL,       0,            1,           -1 },
+	{ "firefox",                    NULL,       NULL,       1 << 3,       0,           1 },
+	{ "Microsoft Teams - Preview",  NULL,       NULL,       1 << 2,       1,           1 },
+	{ "org.remmina.Remmina",        NULL,       NULL,       1 << 1,       1,           1 },
 };
 
 /* layout(s) */
@@ -62,13 +69,33 @@ static const Layout layouts[] = {
 
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
-static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_cyan, "-sf", col_gray4, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", col_gray1, "-nf", col_gray3, "-sb", col_emerald, "-sf", col_gray4, NULL };
+static const char *termcmd[]  = { "gnome-terminal", NULL };
+static const char *lockcmd[]  = { "dwm-screensaver-wrapper", NULL };
+static const char *lovolcmd[]  = { "dwm-audio-wrapper", "dn", NULL };
+static const char *hivolcmd[]  = { "dwm-audio-wrapper", "up", NULL };
+static const char *togglevolcmd[]  = { "dwm-audio-wrapper", "mt", NULL };
+static const char *toggleaudioplaycmd[]  = { "cmus-remote", "-u", NULL };
+static const char *nexttrackcmd[]  = { "cmus-remote", "-n", NULL };
+static const char *prevtrackcmd[]  = { "cmus-remote", "-r", NULL };
+static const char *capturescreencmd[]  = { "dwm-screencapture-wrapper", NULL };
+static const char *showosdcmd[] = { "dwm-osd-wrapper", NULL };
+static const char *passmgrcmd[] = { "dwm-passmgr-wrapper", NULL };
 
-static const Key keys[] = {
+static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
+	{ MODKEY|ControlMask,           XK_a,      spawn,          {.v = passmgrcmd} },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
+	{ MODKEY|ShiftMask,             XK_l,      spawn,          {.v = lockcmd } },
+	{ 0,                XF86XK_AudioPrev,      spawn,          {.v = prevtrackcmd } },
+	{ 0,                XF86XK_AudioNext,      spawn,          {.v = nexttrackcmd } },
+	{ 0,                XF86XK_AudioMute,      spawn,          {.v = togglevolcmd } },
+	{ 0,         XF86XK_AudioLowerVolume,      spawn,          {.v = lovolcmd } },
+	{ 0,         XF86XK_AudioRaiseVolume,      spawn,          {.v = hivolcmd } },
+	{ 0,                XF86XK_AudioPlay,      spawn,          {.v = toggleaudioplaycmd } },
+	{ 0,                   XF86XK_Search,      spawn,          {.v = showosdcmd } },
+	{ 0,                            XK_Print,  spawn,          {.v = capturescreencmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
@@ -90,6 +117,10 @@ static const Key keys[] = {
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	/* AZERTY bindings and tags */
+	{ MODKEY,                       XK_agrave, view,           {.ui = ~0 } },
+	{ MODKEY|ShiftMask,             XK_agrave, tag,            {.ui = ~0 } },
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
@@ -99,7 +130,15 @@ static const Key keys[] = {
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
-	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
+	TAGKEYS(                        XK_ampersand,              0)
+	TAGKEYS(                        XK_eacute,                 1)
+	TAGKEYS(                        XK_quotedbl,               2)
+	TAGKEYS(                        XK_apostrophe,             3)
+	TAGKEYS(                        XK_parenleft,              4)
+	TAGKEYS(                        XK_minus,                  5)
+	TAGKEYS(                        XK_egrave,                 6)
+	TAGKEYS(                        XK_underscore,             7)
+	TAGKEYS(                        XK_ccedilla,               8)
 };
 
 /* button definitions */
diff --git a/config.mk b/config.mk
index 8efca9a..ff270e0 100644
--- a/config.mk
+++ b/config.mk
@@ -4,7 +4,7 @@ VERSION = 6.5
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = /usr/local/stow/dwm-6.4
 MANPREFIX = ${PREFIX}/share/man
 
 X11INC = /usr/X11R6/include
diff --git a/dwm.c b/dwm.c
index c77811f..1c06a01 100644
--- a/dwm.c
+++ b/dwm.c
@@ -36,6 +36,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/XF86keysym.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
-- 
2.34.1

